api/tests/helpers/database.ts

```diff
@@ -1,10 +1,30 @@
 import { beforeEach, afterEach } from 'vitest';
 import prisma from '../../src/database';
 
+let tableNameList: string[] | undefined;
+
+function formatTableNameToPrismaName(tableName: string) {
+  return tableName.charAt(0).toLowerCase() + tableName.slice(1);
+}
+
+const getTableNames = async () => {
+  if (typeof tableNameList !== 'undefined') {
+    return tableNameList;
+  }
+  const tableNames = (
+    await prisma.$queryRaw<{ name: string }[]>`SELECT name FROM sqlite_schema WHERE type ='table' AND name NOT LIKE 'sqlite_%';`
+  );
+  tableNameList = tableNames.map((table) => formatTableNameToPrismaName(table.name));
+  return tableNameList;
+};
+
 export function cleanDatabase() {
   const clean = async () => {
-    await prisma.coreUnit.deleteMany();
-    await prisma.user.deleteMany();
+    const tablenames = await getTableNames();
+    await tablenames.slice(1).reduce(async (acc, tableName) => {
+      const table: any = prisma[tableName as any];
+      return acc.then(() => table.deleteMany());
+    }, (prisma[tablenames[0] as any] as any).deleteMany() as Promise<void>); // can't type properly tablenames received via raw query.
```

Comment:
Why did the [official solution](https://www.prisma.io/docs/concepts/components/prisma-client/crud#deleting-all-data-with-raw-sql--truncate) not work? Can you please bring more context to help you solve it? This code looks like a mess due to:
- multiply raw queries with hardcoded things like `sqlite_` – what happens when we move to postgres?
- constant type re-mapping
- unexplained `slice(1)`
- global variables

If none of the proposed clean versions work, and I can not help you to figure them out, I would prefer to go back to manual deletion (because it cleaner and more readable) than to approve this code